src/global_scheduler/global_scheduler_algorithm.c

```diff
@@ -26,12 +35,39 @@ void handle_task_round_robin(global_scheduler_state *state,
                              global_scheduler_policy_state *policy_state,
                              task *task) {
   CHECKM(utarray_len(state->local_schedulers) > 0,
-         "No local schedulers. We currently don't handle this case.")
-  local_scheduler *scheduler = (local_scheduler *) utarray_eltptr(
-      state->local_schedulers, policy_state->round_robin_index);
-  policy_state->round_robin_index += 1;
-  policy_state->round_robin_index %= utarray_len(state->local_schedulers);
-  assign_task_to_local_scheduler(state, task, scheduler->id);
+         "No local schedulers. We currently don't handle this case.");
+  local_scheduler *ls = NULL;
+  task_spec *task_spec = task_task_spec(task);
+  int i;
+  int num_retries = NUM_RETRIES;
+  bool task_satisfied = false;
+
+  for (i = policy_state->round_robin_index;
+       !task_satisfied && num_retries >= 0;
+       i = (i+1) % utarray_len(state->local_schedulers)) {
+    if (i == policy_state->round_robin_index) {
+      num_retries--;
+    }
+    ls = (local_scheduler *) utarray_eltptr(state->local_schedulers, i);
+    /* Check this local scheduler for hard constraints. */
+    task_satisfied = true;
+    for (int j = 0; j < MAX_RESOURCE_INDEX; j++) {
+      if (ls->info.static_resources[j] < task_required_resource(task_spec, j)) {
+        task_satisfied = false;
+        break; /* Node i doesn't satisfy task's constraint j. Bail. */
+      }
+    }
+  }
+
+  if (task_satisfied) {
+    /* Update next index to try and assign the task. */
+    policy_state->round_robin_index = i; /* i was advanced. */
```

Comment:
It's not a big deal, but if the above goes to the last retry (`num_retries = -1`) and the task is satisfied, I think `round_robin_index` will be set to the same index as the one the task is assigned to, instead of the one after.